marlowe-playground-server/contracts/EscrowWithCollateral.hs

```diff
@@ -0,0 +1,106 @@
+{-# LANGUAGE OverloadedStrings #-}
+module EscrowWithCollateral where
+
+import           Language.Marlowe
+
+main :: IO ()
+main = print . pretty $ contract
+
+
+{- What does the vanilla contract look like?
+  - if Alice and Bob choose
+      - and agree: do it
+      - and disagree: Carol decides
+  - Carol also decides if timeout after one choice has been made;
+  - refund if no choices are made.
+-}
+
+contract :: Contract
+contract = When [Case (Deposit "bob" "bob" ada bobCollateral) contract]
+                10
+                Close
```

Comment:
I think this is an infinite loop, `contract` should probably be `collateralAlice`

Reply:
Lol, I thought haskell interpreter was broken again when I tried to to test Haskell version. Got careless - will fix (the bug not carelessness :) ).

Thanks